OpenSim/Common/FileAdapter.cpp

```diff
@@ -53,27 +53,19 @@ FileAdapter::tokenize(const std::string& str,
 
     std::vector<std::string> tokens{};
 
-    size_type token_start{0}, token_end{0};
-    bool is_token{false};
-    while(token_end < str.length()) {
-        if(delims.find_first_of(str[token_end]) != std::string::npos) {
-            if(is_token) {
-                tokens.push_back(str.substr(token_start, 
-                                            token_end - token_start));
-                is_token = false;
-            }
-        } else {
-            if(!is_token) {
-                token_start = token_end;
-                is_token = true;
-            }
+    size_type token_start{0}, token_end{ std::string::npos };
+    while((token_end = str.find_first_of(delims, token_start)) != std::string::npos) {
+        if (token_end >= token_start) {
```

Comment:
Is it possible for `token_end < token_start`?

Reply:
Yes, because after a token is processed, `token_start = token_end` and immediately thereafter we increment `++token_start`, in which case it is now greater than `token_end`.

Reply:
This if statement also stuck out to me. It seems like it's not needed because of @chrisdembia's documentation, which also says
> If no matches are found, the function returns string::npos.

and `string::npos` is the greatest value of that type.